x-pack/platform/plugins/shared/security/server/session_management/session_index.ts

```diff
@@ -492,8 +497,33 @@ export class SessionIndex {
         indexNeedsRefresh = (await this.bulkDeleteSessions(operations)) || indexNeedsRefresh;
       }
     } catch (err) {
-      logger.error(`Failed to clean up sessions: ${err.message}`);
-      error = err;
+      if (err instanceof errors.ResponseError && err.statusCode === 503) {
+        shardMissingCounter++;
+        if (shardMissingCounter < 10) {
+          logger.warn(
+            `No shards found for session index, skipping session cleanup. This operation has failed ${shardMissingCounter} time(s)`
+          );
+          return {
+            state: {
+              shardMissingCounter,
+            },
+          };
+        }
+
+        const errorMesage = `Failed to clean up sessions: Shards for session index are missing. Cleanup routine has failed ${shardMissingCounter} times. ${getDetailedErrorMessage(
+          err
+        )}`;
+        logger.error(errorMesage);
+        return {
+          error: errorMesage,
```

Comment:
I confirmed and this will report as a failure in the `/api/task_manager/metrics?reset=true` endpoint 👍